pkgs/build-support/rust/fetchcrate.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, fetchurl, unzip }:
+
+{ crateName
+, version
+, sha256
+, ... } @ args:
+
+lib.overrideDerivation (fetchurl ({
+
+  name = "${crateName}-${version}.tar.gz";
+  url = "https://crates.io/api/v1/crates/${crateName}/${version}/download";
+  recursiveHash = true;
+
+  downloadToTemp = true;
+
+  postFetch =
```

Comment:
Is there a reason for not using `fetchzip`? On first glance it looks more applicable here.

Reply:
I was, but for some strange reason, `fetchzip` doesn't work in a sandbox here. I believe this is linked with the fact that curl downloads a file called `download`, which needs to be renamed to `something.tar.gz` for fetchzip to guess how to unzip it.

@Mic92 suggested that solution.